impl/src/planning/subplanners.rs

```diff
@@ -279,23 +301,19 @@ impl<'planner> WorkspaceSubplanner<'planner> {
       }
     }
 
-    let contexts = catalogs
-      .iter()
-      .map(|catalog| {
-        (*catalog)
-          .metadata
-          .resolve
-          .as_ref()
-          .ok_or_else(|| RazeError::Generic("Missing resolve graph".into()))?
-          .nodes
-          .iter()
-          .sorted_by_key(|n| &n.id)
-          .filter_map(|node| self.create_crate_context(node, &package_to_checksum, catalog))
-          .collect::<Result<Vec<CrateContext>>>()
-      })
-      .collect::<Result<Vec<Vec<CrateContext>>>>()?;
+    // If there's no lockfile, the above will miss packages - fill them in here.
+    for entry in &self.crate_catalog.entries {
+      let package = &entry.package;
+      let key = (package.name.to_string(), package.version.clone());
+      if package_to_checksum.get(&key).is_some() {
+        continue;
+      }
+      if let Ok(checksum) = fetch_crate_checksum(&self.settings.index_url, &package.name, &package.version.to_string()) {
```

Comment:
I generally feel this should be an error and we shouldn't be attempting to update the local index to grab a checksum during planning.